files/fluentd/fluentd_prometheus_alerts.yaml

```diff
@@ -31,13 +31,26 @@
     "labels":
       "service": "fluentd"
       "severity": "critical"
-  - "alert": "FluentdErrorsHigh"
-    "annotations":
-      "message": "In the last minute, {{ $value }} errors reported by fluentd {{ $labels.instance }}."
-      "summary": "Fluentd reports high number of errors"
-    "expr": |
-      sum by(instance, job) (rate(fluentd_output_status_num_errors[1m])) > 10
```

Comment:
can you talk a little bit about the difference between the two expressions (this one vs your new one)?

what is `sum by(instance, job)` doing here? (it looks like then we see if we've had > 10 errors in a minute for a minute...)

and instead we are figuring out that if we have 25% of our fluentd records are errors over the course of 5 min for 15 minutes we will alert for high.

Reply:
`sum by(instance, job)` will remove all other labels and sum the resulting datasets that are common. Before this alert was assuming that 10 errors per second was considered a problem. That's a fair belief if fluentd is only outputting 20 logs per second because that would mean that half of the outputs are errors. But the problem is if fluentd is outputting 5000 logs per second. At which point 10 errors per second only represents 0.2% of logs. Perhaps we can play with these percentages a bit and maybe decrease them to 10% or whatever we think might be a good number, however I don't believe that using error count without considering output count is valuable.

Reply:
What value is stored in `$labels.instance` ? Will it help us identify specific node or pod_id?